$NetBSD: patch-src_common.mf,v 1.1 2024/12/08 13:09:43 vins Exp $

* Use standard CFLAGS and LDFLAGS.
* Provide a pkgsrc-compliant install target.

--- src/common.mf.orig	2024-01-21 18:22:22.000000000 +0000
+++ src/common.mf
@@ -28,10 +28,10 @@ wsm_objs = \
 	WmWsmLib/util.o WmWsmLib/utm_send.o
 
 emwm: $(mwm_objs) $(wsm_objs)
-	$(CC) $(LDFLAGS) $(LIBDIRS) -o $@ $(mwm_objs) $(wsm_objs) $(SYSLIBS)
+	$(CC) $(LDFLAGS) -o $@ $(mwm_objs) $(wsm_objs) $(SYSLIBS)
 
 .depend:
-	$(CC) -MM $(INCDIRS) $(mwm_objs:.o=.c) $(wsm_objs:.o=.c) > $@
+	$(CC) -MM $(CFLAGS) $(mwm_objs:.o=.c) $(wsm_objs:.o=.c) > $@
 
 .PHONY: clean install common_install
 
@@ -40,21 +40,11 @@ clean:
 	-rm .depend
 
 common_install:
-	install -m 775 emwm $(PREFIX)/bin/emwm
-	install -m 775 -d $(MANDIR)/man1
-	install -m 664 emwm.1 $(MANDIR)/man1/emwm.1
-	install -m 775 -d $(MANDIR)/man4
-	install -m 664 emwmrc.4 $(MANDIR)/man4/emwmrc.4
-	install -m 775 -d $(RCDIR)
-	install -m 664 system.emwmrc $(RCDIR)/system.emwmrc
-	install -m 775 -d $(APPLRESDIR)
-	install -m 664 Emwm.ad $(APPLRESDIR)/Emwm
-
-uninstall:
-	rm -f $(PREFIX)/bin/emwm
-	rm -f $(MANDIR)/man1/emwm.1
-	rm -f $(MANDIR)/man4/emwmrc.4
-	rm -f $(RCDIR)/system.emwmrc
-	rmdir $(RCDIR)
-	rm -f $(APPLRESDIR)/Emwm
-
+	$(BSD_INSTALL_PROGRAM_DIR) $(DESTDIR)$(BINDIR)
+	$(BSD_INSTALL_PROGRAM) emwm $(DESTDIR)$(BINDIR)
+	$(BSD_INSTALL_MAN_DIR) $(DESTDIR)$(MANDIR)/man1 $(DESTDIR)$(MANDIR)/man4
+	$(BSD_INSTALL_MAN) emwm.1 $(DESTDIR)$(MANDIR)/man1
+	$(BSD_INSTALL_MAN) emwmrc.4 $(DESTDIR)$(MANDIR)/man4
+	$(BSD_INSTALL_DATA_DIR) $(DESTDIR)$(RCDIR) $(DESTDIR)$(APPLRESDIR)
+	$(BSD_INSTALL_DATA) system.emwmrc $(DESTDIR)$(RCDIR)/system.emwmrc
+	$(BSD_INSTALL_MAN) Emwm.ad $(DESTDIR)$(APPLRESDIR)/Emwm
